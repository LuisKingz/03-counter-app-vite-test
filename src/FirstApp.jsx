import PropTypes from 'prop-types';

export const FirstApp = ({title, subTitle, name}) => {    
    return (
    <>
        <h2 data-testid="test-title" >{ title }</h2>
        <p>{ subTitle }</p>
        <p> { name } </p>
    </>
  )
}

// RESTRICCIONES PARA EL TIPO DE DATOS QUE SE RECIBE EN LAS PROPS

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
}

//  VALORES POR DEFECTO EN CASO DE QUE NO SE ENVIEN LOS PROPS. ENTRAN PRIMERO QUE LOS PROPTYPES
FirstApp.defaultProps = {
    name: "No hay nombre",
    subTitle:"No hay Subtitulo"
}