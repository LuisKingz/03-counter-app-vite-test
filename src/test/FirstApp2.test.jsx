import { render,screen } from "@testing-library/react";
import { FirstApp } from "../FirstApp"

describe('pruebas en firstApp', () => { 
    const title = "hola soy un titulo";
    const subTitle = "Hola soy un nuevo subtitulo";

    test('debe de hacer match con el snapshot', () => { 

        const { container } = render(<FirstApp title={ title }  />)
        expect(container).toMatchSnapshot()

     });

     test('debe de mostrar el mensaej "Hola soy un titulo"', () => { 
        render(<FirstApp title={ title } />)
        expect(screen.getByText(title)).toBeTruthy()
      });

      test('debe de mstrar el titulo en un h1', () => { 
        render(<FirstApp title= { title } />)
        expect(screen.getByRole('heading', { level:2 }).innerHTML).toContain(title)
       })

       test('debe de mostrar el subtitulo enviado por props', () => { 
        render(<FirstApp title={ title } subTitle={ subTitle } />)
        //expect(screen.getAllByText(subTitle).length).toBe(2)
        })
 })