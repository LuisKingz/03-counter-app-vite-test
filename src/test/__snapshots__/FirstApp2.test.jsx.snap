// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en firstApp debe de hacer match con el snapshot 1`] = `
<div>
  <h2
    data-testid="test-title"
  >
    hola soy un titulo
  </h2>
  <p>
    No hay Subtitulo
  </p>
  <p>
     
    No hay nombre
     
  </p>
</div>
`;
